src/rules/custom-property-no-empty-line-before/index.js

```diff
@@ -0,0 +1,62 @@
+import {
+  isCustomProperty,
+  optionsHaveException,
+  report,
+  ruleMessages,
+  validateOptions,
+} from "../../utils"
+import _ from "lodash"
+
+export const ruleName = "custom-property-no-empty-line-before"
+
+export const messages = ruleMessages(ruleName, {
+  expected: "Expected single line before custom property",
+  rejected: "Unexpected single line before custom property",
+})
+
+export default function (expectation, options) {
+  return (root, result) => {
+    const validOptions = validateOptions(result, ruleName, {
+      actual: expectation,
+      possible: [
+        true,
+      ],
+    }, {
+      actual: options,
+      possible: {
+        except: ["first-after-non-custom-property-sibling"],
+      },
+      optional: true,
+    })
+    if (!validOptions) { return }
+    const allPropData = []
+    root.walkDecls(decl => {
+      const { prop } = decl
+
+      const propData = prop
+
+      const previousPropData = _.last(allPropData)
+      allPropData.push(propData)
+      if (isCustomProperty(prop)) {
```

Comment:
Instead of a large block within the conditional here, how about an early return?

```js
if (!isCustomProperty(prop)) { return }
// ...
```